src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py

```diff
@@ -33,15 +34,16 @@ class FGSZocaloCallback(CallbackBase):
     """
 
     def __init__(
-        self, parameters: FullParameters, ispyb_handler: FGSISPyBHandlerCallback
+        self, parameters: InternalParameters, ispyb_handler: FGSISPyBHandlerCallback
     ):
         self.grid_position_to_motor_position: Callable[
             [Point3D], Point3D
-        ] = parameters.grid_scan_params.grid_position_to_motor_position
+        ] = parameters.experiment_params.grid_position_to_motor_position
+        self.zocalo_env = parameters.artemis_params.zocalo_environment
         self.processing_start_time = 0.0
         self.processing_time = 0.0
         self.ispyb = ispyb_handler
-        self.zocalo_interactor = ZocaloInteractor(parameters.zocalo_environment)
+        self.zocalo_interactor = ZocaloInteractor(self.zocalo_env)
```

Comment:
Nit: I prefer not holding state we don't need. I think `zocalo_env` can just be a temporary variable in the init